lib/trino-hdfs/src/main/java/io/trino/hdfs/gcs/GoogleGcsConfigurationInitializer.java

```diff
@@ -35,7 +46,23 @@ public GoogleGcsConfigurationInitializer(HiveGcsConfig config)
     {
         config.validate();
         this.useGcsAccessToken = config.isUseGcsAccessToken();
-        this.jsonKeyFilePath = config.getJsonKeyFilePath();
+        this.jsonKeyFilePath = Optional.ofNullable(config.getJsonKey())
+                .map(GoogleGcsConfigurationInitializer::getJsonKeyFilePath)
+                .orElse(config.getJsonKeyFilePath());
+    }
+
+    private static String getJsonKeyFilePath(String jsonKey)
+    {
+        try {
+            // Just create a temporary json key file.
+            Path tempFile = Files.createTempFile("gcs-key-", ".json", PosixFilePermissions.asFileAttribute(EnumSet.of(OWNER_READ, OWNER_WRITE)));
```

Comment:
I am not a big fan of exposing credentials in temporary file - but I assume this is the only way you can integrate with Hive `Configuration`

Reply:
yes, this will be gone after we de-hadoop file systems

obviously this should be used only when deploying in a trusted isolated environment, like k8s